account_payment_order/models/account_move_line.py

```diff
@@ -23,6 +26,25 @@ class AccountMoveLine(models.Model):
         string="Payment lines",
     )
 
+    @api.depends(
+        "move_id", "move_id.invoice_partner_bank_id", "move_id.payment_mode_id"
+    )
+    def _compute_partner_bank_id(self):
+        for ml in self:
+            res = None
+            if (
+                ml.move_id.type in ("in_invoice", "in_refund")
+                and not ml.reconciled
+                and ml.payment_mode_id.payment_order_ok
+                and ml.account_id.internal_type in ("receivable", "payable")
+                and not any(
+                    p_state in ("draft", "open", "generated")
+                    for p_state in ml.payment_line_ids.mapped("state")
+                )
+            ):
+                res = ml.move_id.invoice_partner_bank_id.id
+            ml.partner_bank_id = res
```

Comment:
Anyway, I think you mustn't assign always the bank account, but only when matching the criteria. If not, you are emptying the field in the move line is linked to a payment line for example.

Reply:
done.